aws/bucket.ts

```diff
@@ -110,9 +117,11 @@ export class Bucket extends pulumi.ComponentResource implements cloud.Bucket {
                 err => callback(err, undefined));
         };
 
+        const lambda = callback.createCallbackFunction(name, eventHandler, data, { parent: this });
```

Comment:
Normal pattern is to make the "entryPoint" function that are the toplevel Lambda/FunctionApp entrypoints.  This will call into the javascript function after marshalling the input data properly.  Then, we pass that entrypoint function *along with the* 'data' information to actually go make the final Lambda/FunctionApp.  This gives us the final real Aws/Azure resources with the right entrypoint, and the right user provided configuration (if any).